src/main/java/org/tugraz/sysds/runtime/data/DenseBlockBool.java

```diff
@@ -155,26 +155,31 @@ public DenseBlock set(int r, int c, double v) {
 	
 	@Override
 	public DenseBlock set(DenseBlock db) {
-		System.arraycopy(db.valuesAt(0), 0, _data, 0, _rlen*_odims[0]);
+		double[] data = db.valuesAt(0);
+	    for (int i = 0; i < _rlen*_odims[0]; i++) {
```

Comment:
The formatting is off - please make a full pass over all modified classes in that regard as I'm not commenting on the remaining ones.